uc.c

```diff
@@ -1036,6 +1036,16 @@ uc_err uc_hook_add(uc_engine *uc, uc_hook *hh, int type, void *callback,
         hook->insn = va_arg(valist, int);
         va_end(valist);
 
+
+#ifdef UNICORN_HAS_X86
```

Comment:
in general, we try best to isolate arch-dependent code in uc.c, usually by moving them to qemu/target-xxx/unicorn.c. can you somehow move this code there?